### ISSUE TITLE
view-ajax.php removed in v3.0.17 but still referenced internally and by local_helpdesk

### ISSUE THREAD

**Issue by armando-pradenas** [EduardoKrausME/moodle-local-kopere_dashboard#159]
Hi Eduardo! First off, great work on the v3.0.17 rewrite. While upgrading we noticed a small backward-compatibility issue:

After upgrading to v3.0.17, the user selector in `local_helpdesk` returns a 404 because `view-ajax.php` no longer exists.

**References still pointing to `view-ajax.php`:**

1. `classes/html/data_table.php` (line 484):
   ```php
   "url" => $this->ajaxurl = str_replace("view.php", "view-ajax.php", $this->ajaxurl),
   ```
2. `local_helpdesk` via `url_util::makeurl("users", "load_all_users", [], "view-ajax")`

**Impact:** Any DataTable using server-side processing and any external consumer calling `view-ajax.php` gets a 404 after upgrading to 3.0.17.

**Additional observations:**
- DataTables language strings (`sLengthMenu`, `sSearch`, etc.) render as `[[key,component]]` when the table is initialized inside a dynamic modal (strings not preloaded via `$PAGE->requires->string_for_js`).
- `filter_user.js` in local_helpdesk uses `numberRenderer` for the ID column, displaying IDs with decimals (e.g., `5,00`).

**Workaround:** We created a minimal compatibility shim that handles the `classname=users&method=load_all_users` 
https://github.com/AUTOMATEGY/moodle-local-kopere_dashboard/tree/fix/users-ajax-endpoint-broken

**Comment by marcorossi-altenitalia** [comment]
Hi Eduardo,

We fixed the DataTables language issue in local_helpdesk.

The strings used by local_kopere_dashboard/dataTables_init were not preloaded when the table was initialized inside the dynamic user-selection modal. As a result, labels such as sLengthMenu and sSearch were displayed as:

[[datatables_sLengthMenu,local_kopere_dashboard]]
[[datatables_sSearch,local_kopere_dashboard]]

We fixed this by preloading the required DataTables strings in:

local_helpdesk/classes/util/filter.php

Specifically, the strings are now loaded in filter::load_kopere() using:

```
/**
     * Function load_kopere
     *
     */
    public static function load_kopere() {
        global $PAGE;

        static $loadkopere = false;

        if (!$loadkopere) {
            $PAGE->requires->strings_for_js([
                "datatables_sEmptyTable",
                "datatables_sInfo",
                "datatables_sInfoEmpty",
                "datatables_sInfoFiltered",
                "datatables_sInfoThousands",
                "datatables_sLengthMenu",
                "datatables_sLoadingRecords",
                "datatables_sProcessing",
                "datatables_sErrorMessage",
                "datatables_sZeroRecords",
                "datatables_sSearch",
                "datatables_buttons_print_text",
                "datatables_buttons_copy_text",
                "datatables_buttons_csv_text",
                "datatables_buttons_copySuccess1",
                "datatables_buttons_copySuccess_",
                "datatables_buttons_copyTitle",
                "datatables_buttons_copyKeys",
                "datatables_buttons_select_rows_",
                "datatables_buttons_select_rows1",
                "datatables_buttons_pageLength_",
                "datatables_buttons_pageLength_1",
                "datatables_oPaginate_sNext",
                "datatables_oPaginate_sPrevious",
                "datatables_oPaginate_sFirst",
                "datatables_oPaginate_sLast",
                "datatables_oAria_sSortAscending",
                "datatables_oAria_sSortDescending",
            ], "local_kopere_dashboard");

            $loadkopere = true;
        }
    }

```
This fixes the issue for both the user filter on the Helpdesk index page and the user selector in the ticket form.

The view-ajax.php compatibility shim was also kept as the solution for the broken AJAX endpoint.

Thank you.

**Comment by eduardokraus** [comment]
https://github.com/EduardoKrausME/moodle-local_helpdesk/issues/30
